presentation/details/src/main/java/com/giraffe/details/screens/castDetails/CastDetailsScreen.kt

```diff
@@ -0,0 +1,193 @@
+import androidx.compose.animation.ExperimentalSharedTransitionApi
+import androidx.compose.animation.SharedTransitionLayout
+import androidx.compose.animation.core.animateDpAsState
+import androidx.compose.animation.core.tween
+import androidx.compose.animation.fadeIn
+import androidx.compose.animation.fadeOut
+import androidx.compose.animation.togetherWith
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.systemBarsPadding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import com.giraffe.designsystem.composable.AppBar
+import com.giraffe.designsystem.composable.InfoSection
+import com.giraffe.designsystem.composable.MoviesListSection
+import com.giraffe.designsystem.theme.Theme
+import com.giraffe.details.R
+import com.giraffe.details.components.GallerySection
+import com.giraffe.details.components.MainDetails
+import com.giraffe.details.components.MainDetailsHeader
+import com.giraffe.details.utils.imageSourceToPainter
+import org.koin.androidx.compose.koinViewModel
+
+@Composable
+fun CastDetailsScreen(
+    castDetailsViewModel: CastDetailsViewModel = koinViewModel()
+) {
+    val state by castDetailsViewModel.state.collectAsState()
+    CastDetailsContent(
+        state = state,
+        onYoutubeClick = castDetailsViewModel::onActorYoutubeLinkClicked,
+        onFacebookClick = castDetailsViewModel::onActorFacebookLinkClicked,
+        onInstagramClick = castDetailsViewModel::onActorInstagramLinkClicked,
+        onMoviePosterClick = castDetailsViewModel::onMovieClicked,
+        onShowMoreMoviesClick = castDetailsViewModel::navigateToMoviesListScreen,
+        onShowMoreActorGalleryClick = castDetailsViewModel::navigateToActorGalleryScreen,
+    )
+}
+
+@OptIn(ExperimentalSharedTransitionApi::class)
+@Composable
+fun CastDetailsContent(
+    state: CastDetailsUiState,
+    modifier: Modifier = Modifier,
+    onYoutubeClick: () -> Unit,
+    onFacebookClick: () -> Unit,
+    onInstagramClick: () -> Unit,
+    onMoviePosterClick: (movieId: Int) -> Unit,
+    onShowMoreMoviesClick: () -> Unit,
+    onShowMoreActorGalleryClick: () -> Unit,
```

Comment:
Modifier should be first optional parameter:
```suggestion
    state: CastDetailsUiState,
    onYoutubeClick: () -> Unit,
    onFacebookClick: () -> Unit,
    onInstagramClick: () -> Unit,
    onMoviePosterClick: (movieId: Int) -> Unit,
    onShowMoreMoviesClick: () -> Unit,
    onShowMoreActorGalleryClick: () -> Unit,
    modifier: Modifier = Modifier,
```